reframe/frontend/executors/policies.py

```diff
@@ -138,7 +142,9 @@ def runcase(self, case):
 
             self._pollctl.reset_snooze_time()
             while True:
-                sched.poll(task.check.job)
+                if not task.check.is_dry_run():
+                    sched.poll(task.check.job)
```

Comment:
All the special treatment in the policies is not need, as the `RegressionTest` takes care of dealing with the dry run mode. Only the printout out should be changed here.